src/routes/auth/admin.ts

```diff
@@ -12,35 +13,43 @@ export interface AdminLoginData {
 
 //admin login route
 adminAuthRouter.post("/login", async (req, res) => {
-  const adminLoginData: AdminLoginData = { login: req.body.login, password: req.body.password };
+  try {
+    const adminLoginData: AdminLoginData = { login: req.body.login, password: req.body.password };
 
-  const { valid, errors } = validateAdminLoginData(adminLoginData);
+    const { valid, errors } = validateAdminLoginData(adminLoginData);
 
-  if (!valid) return res.status(400).json(errors);
+    if (!valid) return res.status(400).json(errors);
 
-  // Authenticate user
+    // Authenticate user
 
-  //checking if the admin exists in DB
-  const adminLoginDoc = await AdminModel.findOne({ login: adminLoginData.login }).exec();
-  if (!adminLoginDoc) return res.status(400).json({ login: "email or password is incorrect" });
-  else if (!(await adminLoginDoc.checkPassword(adminLoginData.password))) {
-    return res.status(400).json({ login: "email or password is incorrect" });
-  }
+    //checking if the admin exists in DB
+    const adminLoginDoc = await AdminModel.findOne({ login: adminLoginData.login }).exec();
+    if (!adminLoginDoc) return res.status(400).json({ login: "email or password is incorrect" });
+    else if (!(await adminLoginDoc.checkPassword(adminLoginData.password))) {
+      return res.status(400).json({ login: "email or password is incorrect" });
+    }
 
-  // Return auth token;
-  let accessToken;
-  const TOKEN = process.env.ACCESS_TOKEN_SECRET;
-  TOKEN ? (accessToken = jwt.sign({ login: adminLoginDoc.login }, TOKEN)) : console.log("No secret access token in env");
+    // Return auth token;
+    let accessToken;
+    const TOKEN = process.env.ADMIN_ACCESS_TOKEN_SECRET;
+    TOKEN ? (accessToken = jwt.sign({ login: adminLoginDoc.login }, TOKEN)) : console.log("No secret access token in env");
```

Comment:
With a proper typing file (*.d.ts) you won't have to check for existence of env variable (since we use dotenv-safe).